File: seohwee/week3/A037.py
class Solution:
    def selfDividingNumbers(self, left: int, right: int) -> List[int]:
        sd = []
        
        while left <= right:
            i = left
            
            while i != 0 and i % 10 != 0:
                if left % (i % 10) == 0:
                    i = i // 10
                else:
                    break
            if i == 0:
                sd.append(left)
            left += 1
        return sd

File: SeongKyungKim/week4/A037.java
class Solution {
    public List<Integer> selfDividingNumbers(int left, int right) {
        List<Integer> selfDiv = new ArrayList<>();
        int temp;
        boolean isValid;
        
        for(int i=left; i<=right; i++){
            isValid = true;
            temp = i;
            for(int j=0; j<(int)(Math.log10(i)+1); j++) {
                if(temp%10==0) {
                    isValid = false;
                    break;
                }
                if(i%(temp%10)!=0) {
                    isValid = false;
                    break;
                }
                temp = temp/10;   
            }
            if(isValid) selfDiv.add(i);
         }
        return selfDiv;
      
    }
}
